### PR TITLE
LGA 2678: Initial creation of Email Orchestration API

### PR DESCRIPTION
**Pull request by BenMillar-MOJ**
### [LGA 2678](https://dsdmoj.atlassian.net/jira/software/c/projects/LGA/boards/226?modal=detail&selectedIssue=LGA-2679)

- Adds a base FastAPI app and structure to provide us with a foundation for future development
    - Adds config files for the 3 environments, the env variable ENV is used to select which config to use
    - Adds routing organisation, each endpoint can be created in its own routing file for clean organisation
    - Adds model organisation, models used for expected payloads and responses can be created in their own files
- Adds a default router which routes the user from "/" to "/redoc", this to:
    - 1) Provide an example for how to implement future endpoints 
    - 2) Provide a nicer landing page for the project than a blank page
- Adds unit testing
- Adds Docker support
- Adds documentation for how to start up a local instance of the API and how to set up a docker container of the project

## Any other changes that would benefit highlighting?

I would be extremely grateful if those reviewing could verify they are able to:
- Set up a local instance of the API, and change the port the service runs on
- Run the tests
- Run the API in a docker container
    - Run the tests inside that docker container

This should help test if my documentation is sufficient.

I would greatly value any feedback on the documentation, or the project structure.

Many thanks! 😄 

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,81 @@
...
+
+```
+python --version
+
+pip --version
+```
+
+Ensure both state 'Python 3.11'
+
+#### 4) Update pip and install the required Python dependancies
```
Comment (psweeting1) [inline README.md]: Small typo - *dependencies

File: README.md [hunk 2]
```diff
@@ -1,12 +1,81 @@
...
 
 This is the API used for Email orchestration between CLA services and GOV.UK Notify.
 
+## Development in a virtual environment
+
```
Comment (Pheonnexx) [inline README.md:8]: I am wondering if we should list the python version and other major things that need their version listed at the top in one place so when this changes we dont miss anything.

File: README.md [hunk 3]
```diff
@@ -1,12 +1,81 @@
...
+```
+
+Ensure both state 'Python 3.11'
+
+#### 4) Update pip and install the required Python dependancies
+
+```
+pip install -U pip
+
+pip install -r requirements.txt
```
Comment (Pheonnexx) [inline README.md:46]: Locally this doesn't work for me atm as the file isn't there.

File: README.md [hunk 4]
```diff
@@ -1,12 +1,81 @@
...
+```
+
+#### 5) Launch the service
+
+```
+python start.py
```
Comment (psweeting1) [inline README.md]: `can't open file '/Users/patrick.sweeting/Projects/laa_govuk_notify_orchestrator/start.py': [Errno 2] No such file or directory`. This file isn't in the repo by the looks of it?

File: README.md [hunk 5]
```diff
@@ -1,12 +1,81 @@
...
+```
+
+##### If you want the service to run on a port other than 2500 or a host other than 127.0.0.1
+
+```
+python manage.py --host your.hostname --port
```
Comment (psweeting1) [inline README.md]: Could you provide a port we can use here please, can probably keep to our 8000 area like our other projects.

File: README.md [hunk 6]
```diff
@@ -1,12 +1,81 @@
...
+pip install -U pip
+
+pip install -r requirements.txt
+```
+
+#### 5) Launch the service
+
+```
+python start.py
+```
```
Comment (Pheonnexx) [inline README.md:53]: No such file or directory happening here as well :)

File: README.md (hunk 4 above)
Comment (BenMillar-MOJ) [reply README.md]: This file was renamed to manage.py, apologies

File: README.md (hunk 5 above)
Comment (BenMillar-MOJ) [reply README.md]: Yes, I will do that. The --port argument is optional but good point, will keep the port in the 8000 area

File: README.md (hunk 4 above)
Comment (Pheonnexx) [inline README.md]: No such file or directory :)

File: README.md (hunk 5 above)
Comment (BenMillar-MOJ) [reply README.md]: I have updated the default port to 8026, which is unreserved by default.

File: README.md (hunk 1 above)
Comment (BenMillar-MOJ) [reply README.md]: Fixed this, thank you! 👍

File: README.md (hunk 4 above)
Comment (BenMillar-MOJ) [reply README.md]: This should now be fixed, this was renamed to `python manage.py`.

Many thanks! 👍

**Review by psweeting1 — DISMISSED** [review]
LGTM

**Review by Pheonnexx — APPROVED** [review]
Have done everything requested in the pull request.  LGTM.

**Review by psweeting1 — APPROVED** [review]
LGTM

**Pull request merged**